packages/arb-token-bridge-ui/tests/e2e/specs/urlQueryParam.cy.ts

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
+/* eslint-disable ui-testing/no-hard-wait */
+/**
+ * When user enters the page with query params on URL
+ */
+
+import { formatBigNumber } from '../../../src/util/NumberUtils'
+import { resetSeenTimeStampCache } from '../../support/commands'
+import { getInitialETHBalance, goerliRPC } from '../../support/common'
+
+describe('User enters site with query params on URL', () => {
+  let l1ETHbal: Number
+  // when all of our tests need to run in a logged-in state
+  // we have to make sure we preserve a healthy LocalStorage state
+  // because it is cleared between each `it` cypress test
+  before(() => {
+    getInitialETHBalance(goerliRPC).then(
+      val => (l1ETHbal = parseFloat(formatBigNumber(val, 18, 5)))
+    )
+    // before this spec, make sure the cache is fresh
+    // otherwise pending transactions from last ran specs will leak in this
+    resetSeenTimeStampCache()
+    // log in to metamask before test
+    cy.login('L1')
+    // save state once otherwise `restoreAppState` in beforeEach would clear it
+    cy.saveAppState()
+  })
+  after(() => {
+    // after all assertions are executed, logout and reset the account
+    cy.logout()
+  })
+  beforeEach(() => {
+    cy.restoreAppState()
+  })
+  afterEach(() => {
+    cy.saveAppState()
+  })
+  context('Amount query param', () => {
+    // only ETH is supported for now so by default the following tests are assumed to be ETH
+    it('?amount=max should set transfer panel amount to maximum amount possible based on balance', () => {
+      cy.visit('/', {
+        qs: {
+          amount: 'max'
+        }
+      })
+
+      // wait for input field value to be set
+      // tried with <2500 but the input field value would be 0
+      // because the gas estimation function takes some time
+      cy.wait(2500)
```

Comment:
unfortunately as we don't stub the response and we can't specifically wait for the call with the gas estimation result to return, we could only wait for a set amount of time instead of an interception

do let me know if you find a better way to achieve this :(
happy to change

Reply:
We might want to add https://www.npmjs.com/package/cypress-wait-until if we have multiple use case like this. And we could wait for input change.
But I guess not for now

Reply:
good find!! updated